lib/cylc/remote.py

```diff
@@ -62,91 +67,164 @@ def watch_and_kill(proc):
+        else:
             raise ValueError(
                 r'ERROR: bad cylc executable in global config: %s' % ssh_cylc)
-    command.append(ssh_cylc)
-    command += cmd
+
+    # Insert core raw command after ssh opts and before raw cmd args & opts.
+    command.append(raw_cmd)
+
+    if verbose_on and (
+            cylc.flags.verbose or os.getenv('CYLC_VERBOSE') in
+            ["True", "true"]):
+        command.append(r'--verbose')
+    if debug_on and (
+            cylc.flags.debug or os.getenv('CYLC_DEBUG') in ["True", "true"]):
+        command.append(r'--debug')
+    if args:
+        for arg in args:
+            command.append(quote(arg))
+            # above: args quoted to avoid interpretation by the shell,
+            # e.g. for match patterns such as '.*' on the command line.
+
+    # final ERROR CHECK ?
     if cylc.flags.debug:
-        sys.stderr.write('%s\n' % command)
-    if capture:
-        stdout = PIPE
-    else:
-        stdout = None
-    # CODACY ISSUE:
-    #   subprocess call - check for execution of untrusted input.
-    # REASON IGNORED:
-    #   The command is read from the site/user global config file, but we check
-    #   above that it ends in 'cylc', and in any case the user could execute
-    #   any such command directly via ssh.
+        sys.stderr.write(' '.join(quote(c) for c in command) + '\n')
+
+    return command
+
+
+def remote_cylc_cmd(cmd, user=None, host=None, capture=False, manage=False,
+                    ssh_login_shell=None, ssh_cylc=None, stdin=None):
+    """Run a given cylc command on another account and/or host. AMEND:
+
+    Arguments:
+      * The arguments:
+             cmd --> raw_cmd;
+             user;
+             host;
+             stdin --> set_stdin;
+             ssh_login_shell;
+             ssh_cylc,
+         are passed to 'construct_ssh_cmd, so see associated docstring.
+      * Additionally, specify:
+        capture (boolean):
+            If True, set stdout=PIPE and return the Popen object.
+        manage (boolean):
+            If True, watch ancestor processes and kill command if they change
+            (e.g. kill tail-follow commands when parent ssh connection dies).
+
+    Return:
+        If capture=True, return the Popen object if created successfully.
+        Otherwise, return the exit code of the remote command.
+    """
+    command = construct_ssh_cmd(
+        cmd, user=user, host=host, set_stdin=stdin,
+        ssh_login_shell=ssh_login_shell, ssh_cylc=ssh_cylc)
 
-    proc = Popen(command, stdout=stdout, stdin=stdin)
     if capture:
-        return proc
-    else:
-        if manage:
-            watch_and_kill(proc)
-        res = proc.wait()
-        if WIFSIGNALED(res):
-            sys.stderr.write(
-                'ERROR: remote command terminated by signal %d\n' % res)
-        elif res:
-            sys.stderr.write('ERROR: remote command failed %d\n' % res)
-        return res
+        stdout = PIPE  # Otherwise None by default.
```

Comment:
Need to set `stdout = None` otherwise - or the variable will not be defined for use in the next statement.

Reply:
Sorry Matt, I pushed my recent commits by mistake - they need debugging are definitely not ready for review!